version: '3.8'

services:
  primary:
    image: primary
    build: ./primary/
    ports:
      - "5432:5432"
    volumes:
      - pg_primary_data:/var/lib/postgresql/data
      - ./primary/postgresql.conf:/etc/postgresql/data/postgresql.conf
      - ./primary/pg_hba.conf:/etc/postgresql/data/pg_hba.conf
      - ./primary/backup.sh:/etc/backup.sh
    environment:
      POSTGRES_DB: postgres
      POSTGRES_USER: postgres 
      POSTGRES_PASSWORD: postgres
      PGDATA: /var/lib/postgresql/data
    command: >
      postgres
    networks:
      - pg_network
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U postgres"]
      interval: 5s
      timeout: 5s
      retries: 5

  replica:
    image: replica
    build: ./replica/
    depends_on:
      primary:
        condition: service_healthy
    ports:
      - "5433:5432"
    volumes:
      - pg_replica_data:/var/lib/postgresql/data
      - ./primary/postgresql.conf:/etc/postgresql/data/postgresql.conf
      - ./primary/pg_hba.conf:/etc/postgresql/data/pg_hba.conf
    environment:
      POSTGRES_DB: postgres
      POSTGRES_USER: postgres
      POSTGRES_PASSWORD: postgres
      PGDATA: /var/lib/postgresql/data
    networks:
      - pg_network

volumes:
  pg_primary_data:
  pg_replica_data:

networks:
  pg_network:
    driver: bridge
